src/worker/main.ts

```diff
@@ -127,6 +128,22 @@ const sentryMiddleware = (jobHandler) => async (job) => {
 	}
 };
 
+const setDelayOnRateLimiting = (jobHandler) => async (job) => {
+	try {
+		await jobHandler(job);
+	} catch (err) {
+		if (err instanceof RateLimitingError) {
+			const delay = err.rateLimitReset * 1000 - new Date().getTime();
```

Comment:
I'd probably add a buffer here of a few seconds just to be safe that it doesn't call the API again at the same second that it would allow new requests and to also account for time drift between redis and node